### PR TITLE
simplify `list_data_collate` and `collate_meta_tensor`

### LINKED ISSUE #5917: simplify `list_data_collate` and `collate_meta_tensor ` using `collate`
the generic `collate` API is [implemented in pytorch ](https://github.com/pytorch/pytorch/blob/d7a3f2128fb4457dd60fd5d23e77d2c66a8b0f02/torch/utils/data/_utils/collate.py#L88-L92)

https://github.com/pytorch/pytorch/pull/85748

```
def collate(batch, *, collate_fn_map: Optional[Dict[Union[Type, Tuple[Type, ...]], Callable]] = None):
    r"""
        General collate function that handles collection type of element within each batch
        and opens function registry to deal with specific element types. `default_collate_fn_map`
        provides default collate functions for tensors, numpy arrays, numbers and strings.

```

probably the monai helper functions could be simplified:
https://github.com/Project-MONAI/MONAI/blob/50c3f328cb98e4f0c4e6a38a04a98badf0961c18/monai/data/utils.py#L448-L451

https://github.com/Project-MONAI/MONAI/blob/50c3f328cb98e4f0c4e6a38a04a98badf0961c18/monai/data/utils.py#L427-L430

### PR DESCRIPTION
**Pull request by KumoLiu**
Fixes #5917

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Review by wyli — APPROVED** [review]
thanks, this is a minor refactoring to use the latest pytorch API, please let me know any concerns @ericspod @Nic-Ma.

File: monai/data/utils.py [hunk 1]
```diff
@@ -444,22 +447,28 @@ def pickle_operations(data, key=PICKLE_KEY_SUFFIX, is_encode: bool = True):
     return data
 
 
+def collate_meta_tensor_fn(batch, *, collate_fn_map=None):
+    """collate a sequence of meta tensor into a single batched metatensor"""
```
Comment (ericspod) [inline monai/data/utils.py:451]:
```suggestion
    """
    Collate a sequence of meta tensor into a single batched metatensor. This is called by `collage_meta_tensor` and so should not be used as a collate function directly in dataloaders.
    """
```

I think we should say a little more to make it clear what this function is and to use `collage_meta_tensor` instead.

File: monai/data/utils.py [hunk 2]
```diff
@@ -50,8 +50,11 @@
     issequenceiterable,
     look_up_option,
     optional_import,
+    pytorch_after,
 )
 
+if pytorch_after(1, 13):
+    from torch.utils.data._utils.collate import collate_tensor_fn, default_collate_fn_map
```
Comment (ericspod) [inline monai/data/utils.py:57]: Since the module is `_utils` I wonder if this is meant to be private code we shouldn't rely on. What is the advantage of using these functions over the ones we've been using so far?

File: monai/data/utils.py [hunk 3]
```diff
@@ -479,9 +490,15 @@ def list_data_collate(batch: Sequence):
         Need to use this collate if apply some transforms that can generate batch data.
 
     """
+
+    if pytorch_after(1, 13):
```
[the remark this answers is not in the document]
Comment (KumoLiu) [reply monai/data/utils.py:494]: Yes, you are right. But it needs to go here to avoid circular import.

File: monai/data/utils.py (hunk 2 above)
Comment (KumoLiu) [reply monai/data/utils.py:57]: The benefits may be we could simplify L463-L466 after we add a new collate map for `MetaTensor`.
https://github.com/Project-MONAI/MONAI/blob/85243f52085b0946870fc7808c484c8b639eaf1b/monai/data/utils.py#L463-L466

Comment (ericspod) [reply monai/data/utils.py:57]: I would just be careful using `_utils` directly. [Here](https://github.com/pytorch/pytorch/blob/d7a3f2128fb4457dd60fd5d23e77d2c66a8b0f02/torch/utils/data/dataloader.py#L64C37-L64C37) the function is imported into a public module. It seems `torch.utils.data.dataloader.default_collate` is meant to be the public member which just happens to be implemented as an alias to a private one, but this could change without breaking the public interface.

Comment (KumoLiu) [reply monai/data/utils.py:57]: Yes, we should always be careful with the private code. I didn't find an alias for these two. 
But it seems they tend to extend collate functions that can register collate functions to handle specific types.
https://github.com/pytorch/pytorch/pull/85748

Do you have some suggestions for using these or we could just leave them here?

Comment (ericspod) [reply monai/data/utils.py:57]: It seems that `default_collate_fn_map` is meant to be internal and used only by `default_collate`, but `collate` is also private so shouldn't be used either? We have no way of tweaking collate behaviour if so. I wouldn't myself play around with Pytorch internals like this but there's no other way it seems to get your behaviour and reuse things. Let's put in comments where you do the imports stating you're importing private code for reuse purposes so we keep track in case things break in the future.

Comment (KumoLiu) [reply monai/data/utils.py:58]: Sure, thanks! I added comments in the latest commit.
